pkg/gather/clusterconfig/openshift_apiserver_operator_logs.go

```diff
@@ -0,0 +1,112 @@
+package clusterconfig
+
+import (
+	"bufio"
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"strings"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatherOpenShiftAPIServerOperatorLogs collects logs from openshift-apiserver-operator with following substrings:
+//   - "the server has received too many requests and has asked us"
+//   - "because serving request timed out and response had been started"
+//
+// The Kubernetes API https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/pod_expansion.go#L48
+// Response see https://docs.openshift.com/container-platform/4.6/rest_api/workloads_apis/pod-core-v1.html#apiv1namespacesnamespacepodsnamelog
+//
+// Location in archive: logs/openshift-api-server-operator
+func GatherOpenShiftAPIServerOperatorLogs(g *Gatherer) func() ([]record.Record, []error) {
+	return func() ([]record.Record, []error) {
+		gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		client := gatherKubeClient.CoreV1()
+
+		records, err := gatherOpenShiftAPIServerOperatorLogs(g.ctx, client)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		return records, nil
+	}
+}
+
+func gatherOpenShiftAPIServerOperatorLogs(ctx context.Context, coreClient corev1client.CoreV1Interface) ([]record.Record, error) {
+	const namespace = "openshift-apiserver-operator"
+	var (
+		messagesToSearch = []string{
+			"the server has received too many requests and has asked us",
+			"because serving request timed out and response had been started",
+		}
+		sinceSeconds int64 = 86400     // last day
+		limitBytes   int64 = 1024 * 64 // maximum 64 kb of logs
+	)
+
+	pods, err := coreClient.Pods(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var records []record.Record
+
+	for _, pod := range pods.Items {
+		request := coreClient.Pods(namespace).GetLogs(pod.Name, &corev1.PodLogOptions{
+			SinceSeconds: &sinceSeconds,
+			LimitBytes:   &limitBytes,
+		})
+
+		logs, err := filterLogs(ctx, request, messagesToSearch)
+		if err != nil {
+			return nil, err
+		}
+
+		records = append(records, record.Record{
+			Name:        "logs/openshift-api-server-operator",
```

Comment:
I think we already store some pod logs elsewhere in the archive (I think in case of unhealthy pods). I would suggest to use the same or similar dir. What others think?

Reply:
It would be better to also include the pod name.

Reply:
See https://github.com/openshift/insights-operator/blob/master/pkg/gather/clusterconfig/operators.go#L164. We may already have the log, but we don't have to so the question is how to handle these situations.

Reply:
It seems strange to have logs under "config" folder, but if we already store logs somewhere, then, sure, I'll change it :)